cfme/cloud/instance/__init__.py

```diff
@@ -284,9 +284,10 @@ def find_quadicon(self, **kwargs):
         Args:
         Returns: entity of appropriate type
         """
-        view = navigate_to(self.parent, 'All')
-        view.toolbar.view_selector.select('Grid View')
-
+        archived = kwargs.pop('archived', None)
+        if not archived:
+            view = navigate_to(self.parent, 'All')
+        self.view.toolbar.view_selector.select('Grid View')
```

Comment:
This block doesn't look right at all to me.

Setting a local `view`, but then referencing `self.view` instead?
Only navigating when `archived` kwarg is not passed?

Reply:
Consider actually using view entities directly in the test, as the TODO in the docblock mentions, instead of trying to further expand what its doing.